Sources/NIOSSL/NIOSSLHandler.swift

```diff
@@ -200,20 +255,70 @@ public class NIOSSLHandler : ChannelInboundHandler, ChannelOutboundHandler, Remo
         case .idle, .handshaking, .additionalVerification:
             // we should not flush immediately as we have not completed the handshake and instead buffer the flush
             self.bufferFlush()
-        case .active, .unwrapping, .closing, .unwrapped, .closed:
+        case .active, .unwrapping, .closingOutput, .closing, .unwrapped, .inputClosed, .outputClosed, .closed: // TODO: .closingOutput + .inputClosed correct here?
             self.bufferFlush()
             self.doUnbufferWrites(context: context)
         }
     }
-    
+
     public func close(context: ChannelHandlerContext, mode: CloseMode, promise: EventLoopPromise<Void>?) {
-        guard mode == .all else {
-            // TODO: Support also other modes ?
+        switch mode {
+        case .output:
+            self.closeOutput(context: context, promise: promise)
+        case .all:
+            self.closeAll(context: context, promise: promise)
+        default:
```

Comment:
May as well write `case .input` as it's the only case left and take advantage of the exhaustive switch.